preview/MsixCore/MsixCoreInstaller/FileTypeAssociation.cpp

```diff
@@ -304,6 +282,13 @@ HRESULT FileTypeAssociation::ProcessFtaForAdd(Fta& fta)
         RETURN_IF_FAILED(verbCommandKey.SetStringValue(L"", verbCommand));
     }
 
+    for (auto extensionName = fta.extensions.begin(); extensionName != fta.extensions.end(); ++extensionName)
+    {
+        m_msixRequest->GetRegistryDevirtualizer()->DeleteKeyIfPresent(rootHive, classesKeyPath.c_str(), *extensionName);
+    }
+
+    m_msixRequest->GetRegistryDevirtualizer()->DeleteKeyIfPresent(rootHive, classesKeyPath.c_str(), fta.progID);

```

Comment:
>progID [](start = 104, length = 6)

the progID that we create here may not be what's in the virtual hive. This needs to be read off of the virtual key. i.e. this is a author-specified arbitrary string.

for instance in VLC, .mp4 key says it's associated with VLC.mp4 (the progID) so they've created a VLC.mp4 key that specifies the information.
our installer will associate .mp4 with MsixCoreVLC.mp4, so we've created a MsixCoreVLC.mp4 key that specifies the information.

Reply:
Modified to get the default key value(progId) for an extension in a FTA. Then passed this over to DeleteKeyIfPresent to delete it from either the user or machine locations in the virtual mount.